### PR TITLE
SystemCommand: Annotate system_command and its derivatives.

### PR DESCRIPTION
**Pull request by Kentzo**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by Kentzo** [comment]
References https://github.com/Homebrew/brew/pull/14629

File: Library/Homebrew/cask/url.rb [hunk 1]
```diff
@@ -102,7 +102,7 @@ def call
           result = curl_output("--fail", "--silent", "--location", @uri)
           result.assert_success!
 
-          page = result.stdout
+          page = T.cast(result.stdout, T.all(String, PageWithURL))
```
Comment (reitermarkus) [inline Library/Homebrew/cask/url.rb:105]: The cast should be after the `extend` below.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @Kentzo! Could you explain a little bit about how you ran into this issue?

**Comment by Kentzo** [comment]
The lack of annotations for `system_command` caused one of the tests to fail in my sudo PR because sorbet saw a type cast from `Array` to `untyped`.Adding an annotation snowballed quickly into a larger diff.Best RegardsIlya KulakovOn Apr 29, 2023, at 3:16 AM, Mike McQuaid ***@***.***> wrote:﻿
Thanks @Kentzo! Could you explain a little bit about how you ran into this issue?

File: Library/Homebrew/cask/url.rb [hunk 2]
```diff
@@ -102,7 +102,7 @@ def call
...
           page.extend PageWithURL
```
Comment (dduugg) [inline Library/Homebrew/cask/url.rb:106]: Rarely have I seen a module so clearly named to be a tuple. How much work is it to refactor as a T::Struct consisting of page and url? (I assume the string being decorated is the page?)

Comment (dduugg) [reply Library/Homebrew/cask/url.rb:106]: Something like this, as a starting point: https://github.com/Homebrew/brew/commit/8adce0f8e0dc2e192fcb8fa489f438fb4254fbf5

**Comment by reitermarkus** [comment]
I have opened https://github.com/sorbet/sorbet/issues/6984, since having the shorthand

```ruby
out, err, status = system_command! "true"
```

is much more convenient to use and it mirrors `Open3.capture3`.

**Pull request closed without merging**